src/core/RDP/nla/credssp.hpp

```diff
@@ -197,6 +213,33 @@ namespace BER {
         return head;
     }
 
+    inline std::vector<uint8_t> mkOidHeader(uint32_t payload_size)
+    {
+        std::vector<uint8_t> head;
+        backward_push_oid_header(head, payload_size);
+        std::reverse(head.begin(), head.end());
+        return head;
+    }
+
+    inline std::vector<uint8_t> mkOid(bytes_view oid)
+    {
+        std::vector<uint8_t> ret = mkOidHeader(oid.size());
+        ret << std::vector<uint8_t>(oid.data(), oid.data()+oid.size());
```

Comment:
`<<` should take a `bytes_view`/`array_view`. I also think `<<=` to be more explicit than `<<`.

But is it really better than `insert()` ?
```cpp
ret.insert(ret.end(), oid.begin(), oid.end());
```